checkbox-support/checkbox_support/scripts/image_checker.py

```diff
@@ -7,16 +7,42 @@
 import argparse
 from os.path import exists
 from checkbox_support.snap_utils.system import on_ubuntucore
+import shutil
+from subprocess import PIPE, run
 
 
-def get_type():
+def get_type() -> str:
     """
     Return the type of image.
     """
     return "core" if on_ubuntucore() else "classic"
 
 
-def get_source():
+def has_desktop_environment() -> bool:
+    """
+    Returns whether there's a desktop environment
+    """
+    if not shutil.which("dpkg") or on_ubuntucore():
+        # core doesn't have dpkg
+        return False
+
+    # if we found any of these packages, we are on desktop
+    if (
+        run(
+            ["dpkg", "-l", "ubuntu-desktop"], stdout=PIPE, stderr=PIPE
+        ).returncode
+        == 0
+        or run(
+            ["dpkg", "-l", "ubuntu-desktop-minimal"], stdout=PIPE, stderr=PIPE
+        ).returncode
+        == 0
+    ):
+        return True
+
+    return False
```

Comment:
```suggestion
    packages = ["ubuntu-desktop", "ubuntu-desktop-minimal"]
    for package in packages:
        result = run(["dpkg", "-l", package], stdout=DEVNULL, stderr=DEVNULL)
        if result.returncode == 0:
            return True
    return False
```

Here are some suggestions that you can take
- Use `DEVNULL` instead of `PIPE` since we don't care ant output.
- Return function once we get the positive outcome